nifi-nar-bundles/nifi-azure-bundle/nifi-azure-reporting-task/src/main/java/org/apache/nifi/reporting/azure/loganalytics/AzureLogAnalyticsReportingTask.java

```diff
@@ -0,0 +1,335 @@
+
+    private static final String JVM_JOB_NAME = "jvm_global";
+    private static final Charset            UTF8                = Charset.forName("UTF-8");
+    private static final String             HMAC_SHA256_ALG     = "HmacSHA256";
+    static final DateTimeFormatter   RFC_1123_DATE_TIME  = DateTimeFormatter.ofPattern("EEE, dd MMM yyyy HH:mm:ss O");
+    private volatile JvmMetrics virtualMachineMetrics;
+
+
+    static final PropertyDescriptor LOG_ANALYTICS_WORKSPACE_ID = new PropertyDescriptor.Builder()
+            .name("Log Analytics Workspace Id")
+            .description("Log Analytics Workspace Id")
+            .required(true)
+            .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .sensitive(true)
+            .build();
+    static final PropertyDescriptor LOG_ANALYTICS_CUSTOM_LOG_NAME = new PropertyDescriptor.Builder()
+            .name("Log Analytics Custom Log Name")
+            .description("Log Analytics Custom Log Name")
+            .required(false)
+            .defaultValue("nifimetrics")
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
+            .build();
+    static final PropertyDescriptor LINUX_PRIMARY_KEY = new PropertyDescriptor.Builder()
+            .name("Linux Primary Key")
+            .description("Connected Sources - Linux Primary Key")
+            .required(true)
+            .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .sensitive(true)
+            .build();
+    static final PropertyDescriptor APPLICATION_ID = new PropertyDescriptor.Builder()
+            .name("Application ID")
+            .description("The Application ID to be included in the metrics sent to Azure Log Analytics WS")
+            .required(true)
+            .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
+            .defaultValue("nifi")
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+    static final PropertyDescriptor INSTANCE_ID = new PropertyDescriptor.Builder()
+            .name("Instance ID")
+            .description("Id of this NiFi instance to be included in the metrics sent to Azure Log Analytics WS")
+            .required(true)
+            .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
+            .defaultValue("${hostname(true)}")
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+    static final PropertyDescriptor PROCESS_GROUP_IDS = new PropertyDescriptor.Builder()
+            .name("Process group ID(s)")
+            .description("If specified, the reporting task will send metrics the configured ProcessGroup(s) only. Multiple IDs should be separated by a comma. If"
```

Comment:
It's best to keep consistent with verbiage. Here we have "Process group", "ProcessGroup", "group-IDs". Recommend "Process Group" / "Process Group IDs". Rather than "NiFi-Flow-ProcessGroup" I think you mean the "Root Process Group"

Reply:
We want to get process group id or comma separate process group id list.

Reply:
Oh. I get your point. I mean the description changes (NiFi-Flow-ProcessGroup-> Root Process Group).